modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultMuleApplication.java

```diff
@@ -244,7 +244,8 @@ private void doInit(boolean lazy, boolean disableXmlValidations) {
                                                                     descriptor.getDeploymentProperties())))
               .setPolicyProvider(policyManager)
               .setRuntimeLockFactory(runtimeLockFactory)
-              .setMemoryManagementService(memoryManagementService);
+              .setMemoryManagementService(memoryManagementService)
+              .setArtifactCoordinates(descriptor.getBundleDescriptor());
```

Comment:
This should also be done in `DefaultApplicationPolicyInstance#initPolicyContext` and `DefaultMuleDomain#doInit`